test/integ/test_cursor.py

```diff
@@ -1318,6 +1318,34 @@ def test_resultbatch_lazy_fetching(conn_cnx, result_format, patch_path):
                 assert patched_download.call_count == 5
 
 
+@pytest.mark.skipolddriver(reason="new feature in v2.5.0")
+@pytest.mark.parametrize(
+    "result_format,patch_path",
+    (
+        ("json", "snowflake.connector.result_batch.JSONResultBatch.create_iter"),
+        ("arrow", "snowflake.connector.result_batch.ArrowResultBatch.create_iter"),
+    ),
+)
+def test_resultbatches_can_access_schema(conn_cnx, result_format, patch_path):
+    rowcount = 1000000
+    with conn_cnx(
+        session_parameters={"python_connector_query_result_format": result_format}
+    ) as con:
+        with con.cursor() as cur:
+            cur.execute(
+                f"select seq4() as c1, randstr(1,random()) as c2 "
+                f"from table(generator(rowcount => {rowcount}));"
+            )
+            result_batches = cur.get_result_batches()
+            batch_schemas = [batch._schema for batch in result_batches]
```

Comment:
_schema is the exposed interface so it should be a public attribute. A '_' prefix is not needed here. Please make sure when there is no result(row count is 0) we still have the schema.

Reply:
Sounds good. I will add a test for this and update the function accordingly.

Reply:
Updated `_schema` to `schema` and added the test